airflow/providers/google/cloud/transfers/sftp_to_gcs.py

```diff
@@ -168,6 +181,32 @@ def _copy_single_object(
             self.log.info("Executing delete of %s", source_path)
             sftp_hook.delete_file(source_path)
 
+    def _stream_single_object(self, sftp_hook: SFTPHook, gcs_hook: GCSHook, source_path: str, destination_object: str):
+        """Helper function to stream single object."""
+        self.log.info(
+            "Executing stream of %s to gs://%s/%s",
+            source_path,
+            self.destination_bucket,
+            destination_object,
+        )
+
+        client = gcs_hook.get_conn()
+        dest_blob = client.bucket(self.destination_bucket).blob(destination_object)
+
+        with sftp_hook.get_conn().file(source_path, 'rb') as source_stream:
+            if source_stream_wrapper: 
+                source_stream = source_stream_wrapper(source_stream)
+            with dest_blob.open("wb") as write_stream:
+                while True:
+                    chunk = source_stream.read(self.stream_chunk_size)
+                    if not chunk:
+                        break
+                    write_stream.write(chunk)
```

Comment:
Is there a specific reason you use dest_blob.open instead of streaming it with blob instead of something like:
`with sftp_hook.get_conn().file(source_path, 'rb', self.stream_chunk_size) as source_stream:
    dest_blob.upload_from_file(source_stream)
`

This code may not be exact, but I know blob has an upload_from_file that takes a file-like object, open for reading, and uploads its contents.